File: python/6kyu/Take_a_Ten_Minutes_Walk.py
def is_valid_walk(walk):
    if len(walk) > 10 or len(walk) < 10:
        return False
    
    ns = 0
    we = 0
    
    for dir in walk:
        if dir == "n":
            ns += 1
        if dir == "s":
            ns -= 1
        if dir == "w":
            we += 1
        if dir == "e":
            we -= 1
    return len(walk) == 10 and ns == 0 and we == 0

File: typescript/6kyu/Take_a_Ten_Minutes_Walk.ts
export const isValidWalk = (walk: string[]): boolean => {
  if (walk.length > 10 || walk.length < 10) return false;
  let ns = 0;
  let we = 0;
  for (let dir of walk) {
    if (dir === "n") ns += 1;
    if (dir === "s") ns -= 1;
    if (dir === "w") we += 1;
    if (dir === "e") we -= 1;
  }

  return walk.length == 10 && ns === 0 && we === 0;
};
